File: src/main/java/com/github/weewar/mapviewer/utils/Images.java
package com.github.weewar.mapviewer.utils;

import com.github.weewar.mapviewer.exceptions.ImageIOException;

import javax.imageio.ImageIO;
import java.awt.image.RenderedImage;
import java.io.ByteArrayOutputStream;
import java.io.IOException;

public class Images {
    public static byte[] toPNG(RenderedImage image) throws ImageIOException {
        try {
            ByteArrayOutputStream imageData = new ByteArrayOutputStream();
            ImageIO.write(image, "png", imageData);
            return imageData.toByteArray();
        } catch (IOException e) {
            throw new ImageIOException(e);
        }
    }
}

File: src/main/kotlin/com/github/weewar/mapviewer/utils/Images.kt
package com.github.weewar.mapviewer.utils

import com.github.weewar.mapviewer.model.ImageIOException
import java.awt.image.RenderedImage
import java.io.ByteArrayOutputStream
import java.io.IOException
import javax.imageio.ImageIO

object Images {
    @Throws(ImageIOException::class)
    fun toPNG(image: RenderedImage): ByteArray =
            try {
                val imageData = ByteArrayOutputStream()
                ImageIO.write(image, "png", imageData)
                imageData.toByteArray()
            } catch (e: IOException) {
                throw ImageIOException(e)
            }
}
